project_setup_best_practices.md

```diff
@@ -0,0 +1,180 @@
+utility files they are testing. The name of a test file should follow the
+pattern of `*_test.py`. For example, the tests for
+`/keras_tuner/engine/hyperparameters.py` are in
+`/keras_tuner/engine/hyperparameters_tests.py`.
+
+Integration tests may be contained in their own `/keras_tuner/integration_tests`
+directory, as they may require extra files such as data.
+
+While our unit test placement is not suggested in the
+[good practices of pytest](https://docs.pytest.org/en/6.2.x/goodpractices.html)
+doc, we recommend this approach to improve the discoverability of the unit
+tests for new contributors. This discoverability doubles up as a method of
+documentation; when users want to see what `util.utility_function()` does, they
+can simply open the conveniently located sibling file, `util_test.py`.
+
+### Test Coverage
+
+We use [CodeCov](https://about.codecov.io/) to track the test coverage.You may
+also refer to
+[these settings](https://github.com/keras-team/keras-tuner/blob/1.1.0/setup.cfg#L24-L28)
+in `setup.cfg`. We will see more about it in the continuous integration section.
+
+Pytest CodeCov supports a wildcard exclude field, which should be set to
+include `*_test.py`, as to ensure that tests are not included in the code
+coverage count.
+
+### Useful code snippets
+Fix the random seed for all tests:
+[Link1](https://github.com/keras-team/keras-tuner/blob/1.1.0/tests/conftest.py#L8-L17),
+[Link2](https://github.com/keras-team/keras-tuner/blob/master/tests/unit_tests/randomness_test.py),
+[Link3](https://www.tensorflow.org/api_docs/python/tf/keras/utils/set_random_seed).
+
+Create a temporary path for testing: [Link](https://docs.pytest.org/en/6.2.x/tmpdir.html).
+
+## Code styles
+
+### Importing Keras modules
+
+For projects based on Keras and TensorFlow, top-level imports are encouraged, like
+shows in the following example.
+
+```py
+import tensorflow as tf
+from tensorflow import keras
+```
+
+Exceptions may be acceptable when the module appeared too many times in the code,
+like `keras.layers`.
+
+### Linting and formatting
+
+We use
+[black](https://black.readthedocs.io/en/stable/),
+[isort](https://pycqa.github.io/isort/), 
+[flake8](https://flake8.pycqa.org/en/latest/)
+to lint and format the code. black is to generally format the code. isort is to
+sort the imports. flake8 is for some additional checks that black doesn't do,
+like the long lines with a single string. You can see the relevant sections of
+[setup.cfg](https://github.com/keras-team/keras-tuner/blob/1.1.0/setup.cfg) for
+the detailed configuration of these tools.
+
+The user does not need to know how to use these tools to lint or format the
+code. We provide them with two shell scripts:
+[`/shell/lint.sh`](https://github.com/keras-team/keras-tuner/blob/master/shell/lint.sh)
+and
+[`/shell/format.sh`](https://github.com/keras-team/keras-tuner/blob/master/shell/format.sh).
+In these scripts, we also check and add the Apache 2.0 License head to every
+file.
+
```

Comment:
Are these scripts globals for the Keras ecosystem or do we have a baseline template that need to be customized for every repository? Also if we are going to customize this scripts for every repository they will go naturally out of sync over time if we don't handle a master.

We partially discussed this mentioning https://stackoverflow.com/questions/59047996/how-can-i-configure-pre-commit-tool-to-rely-on-a-single-location-for-all-hook-co

But as currently we don't want to have automatic pre-commit hooks checks before commits we could probably evaluate to set a specif repository collecting `master templates` of these common infra scripts.